### ISSUE TITLE
font conver

### ISSUE THREAD

**Issue by ihuoxin** [lvgl/lv_font_conv#66]
### Perform all steps below and tick them with [x]

- [x] Check the related part of the [Documentation](https://docs.lvgl.io/)
- [x] Update lvgl to the latest version
- [x] Reproduce the issue in a [Simulator](https://docs.lvgl.io/latest/en/html/get-started/pc-simulator.html)

### Describe the bug

​	    Garbled characters in the offline font conversion tool

1. The converted output_roboto_r.c internal font display is normal, and the converted output_roboto_r.font external font display shows garbled characters;
2. problem analysis:
   Use lvgl\tests\lv_test_core\lv_test_font_loader.c to compare the internal and external font libraries between the output_roboto_r.c file and output_roboto_r.font file;
   - It is analyzed that the adv_w parameter has different values in the output_roboto_r.c and output_roboto_r.font files;
3. Convert output_roboto_r.c internal font command:
   lv_font_conv --font Roboto-Regular.ttf -r 0x20-0x7F -r 0x401,0x410-0x44F,0x451 --size 16 --format lvgl --bpp 4 --no-compress -o output_roboto_r.c
4. Convert output_roboto_r.font external font library command:
   lv_font_conv --font Roboto-Regular.ttf -r 0x20-0x7F -r 0x401,0x410-0x44F,0x451 --size 16 --format bin --bpp 4 --no-compress -o output_roboto_r.font
5. See the attachment for the converted font file;
6. Please help analyze what caused the converted output_roboto_r.font to display garbled characters;

### To Reproduce

1. Convert output_roboto_r.c and output_roboto_r.font through the font conversion command, and compare them with the lv_test_font_loader tool;

### Expected behavior

1. It is expected that the converted external font library can be displayed normally;

### Screenshots or video
1. Use the lv_test_font_loader tool to compare adv_w screenshots with different values for internal and external fonts.

![scr_font](https://user-images.githubusercontent.com/23013817/124415419-6a610a00-dd87-11eb-927d-47adb679baea.png)

2.See the attachment for the converted font file.
 
[converted_font_file.zip](https://github.com/lvgl/lvgl/files/6761515/converted_font_file.zip)

**Comment by puzrin** [comment]
@kisvegabor could you rephrase problem description in short (how to reproduce, is that regression and so on)?

**Comment by kisvegabor** [comment]
Run these 2 commands using the font from [Roboto-Regular.ttf.zip](https://github.com/lvgl/lv_font_conv/files/6847122/Roboto-Regular.ttf.zip)

```
lv_font_conv --font Roboto-Regular.ttf -r 0x20-0x7F -r 0x401,0x410-0x44F,0x451 --size 16 --format lvgl --bpp 4 --no-compress -o output_roboto_r.c
```

```
lv_font_conv --font Roboto-Regular.ttf -r 0x20-0x7F -r 0x401,0x410-0x44F,0x451 --size 16 --format bin --bpp 4 --no-compress -o output_roboto_r.font
```

I think the main issue is `advance_width_format` is `0` for binary format but `1` for `lvgl` format.

I don't know if it's a regression.

**Comment by puzrin** [comment]
1. https://github.com/choffmeister/roboto-fontface-bower/tree/master/fonts/roboto - this source produces advance_width_format = 1.
2. https://github.com/lvgl/lv_font_conv/blob/master/lib/font/font.js#L43 - i suppose attached font has no kerning.
3. `--format lvgl` is cropped and has no `advance_width_format` support.

So, my question is - what bug do you try to report, and why do you think this is convertor bug at all?

**Comment by kisvegabor** [comment]
With [Roboto-Medium.woff.zip](https://github.com/lvgl/lv_font_conv/files/6854020/Roboto-Medium.woff.zip)
there is no difference in `adv_w`. 

@ihuoxin Can you check it too?

@puzrin In the lvgl format `adv_w` is always represented as a fixed-point number. The issue was that with the ttf font `adv_w` was 71 (fractional) for a glyph in lvgl format and 4*16 (not fractional) for bin format. So the output is different in the two formats.

**Comment by puzrin** [comment]
@puzrin 

> So the output is different in the two formats.

I still don't understand why this should be issue in this tracker. `--format lvgl` has less features, and it MUST be different when something advanced used in `--format bin`.

outputs correctness != outputs equality.

**Comment by kisvegabor** [comment]
I see but if `adv_w` is known with higher resolution (fixed-point representation) why is it truncated in the bin format (setting `advance_width_format=0`)?

**Comment by puzrin** [comment]
> I see but if `adv_w` is known with higher resolution (fixed-point representation) why is it truncated in the bin format (setting `advance_width_format=0`)?

To save some space. Because high resolution without kerning is useless. I don't remember details.

**Comment by kisvegabor** [comment]
If so, IMO, rounded values (e.g. 71->64) should be used in the lvgl format too for compatibility.

**Comment by puzrin** [comment]
I don't like idea to spend [my] time for text format, except bugfixes (when something not working at all). It's technically impossible to make it strict equal to binary anyway. May be worth to "let text format die" and wait until been replaced with binary?

**Comment by kisvegabor** [comment]
Ok, as the woff format works well let's close this issue.
